In [1]:
#Para manejos de directorios y archivos
import os

# Computacion numerica en Python, como arrays y matrices
import numpy as np

#Proporciona estrucutra de datos y herramientas para el analisis de datos
import pandas as pd

#Datos en una tabla
from tabulate import tabulate

#Para graficación, visulaizacion de datos
from matplotlib import pyplot
#Superficies en 3D
from mpl_toolkits.mplot3d import Axes3D

#Para separar el 20% y 80%
from sklearn.model_selection import train_test_split

#Modulo de optimizacion en scipy
from scipy import optimize
from scipy.io import loadmat

import torch
from torch import optim  # Para optimizadores como SGD, Adam, etc.
from torch import nn  # Todos los módulos de redes neuronales
from torch.utils.data import DataLoader  # Facilita la gestión de conjuntos de datos creando mini lotes, etc.

import torchvision  # Paquete torch para cosas relacionadas con la visión
import torchvision.transforms as transforms  # Transformaciones que podemos realizar en nuestro conjunto de datos para aumentar
import torch.nn.functional as F  # Funciones sin parámetros, como (algunas) funciones de activación
import torchvision.datasets as datasets  # Conjuntos de datos estándar
from tqdm import tqdm  # ¡Para una barra de progreso agradable!
from torch.utils.data import Dataset

#Visualizar graficas en linea
%matplotlib inline

In [2]:
from google.colab import drive
drive.mount("/content/gdrive")

Mounted at /content/gdrive


Red Neuronal (Pytorch)

In [3]:
dataset = pd.read_csv('/content/gdrive/MyDrive/SIS 420 INTELIGENCIA ARTIFICAL/datasets/covtype.csv', delimiter=',')
dataset

Elevation  Aspect  Slope  Horizontal_Distance_To_Hydrology  \
0            2596      51      3                               258   
1            2590      56      2                               212   
2            2804     139      9                               268   
3            2785     155     18                               242   
4            2595      45      2                               153   
...           ...     ...    ...                               ...   
581007       2396     153     20                                85   
581008       2391     152     19                                67   
581009       2386     159     17                                60   
581010       2384     170     15                                60   
581011       2383     165     13                                60   

        Vertical_Distance_To_Hydrology  Horizontal_Distance_To_Roadways  \
0                                    0                              510   
1                                   -6                              390   
2                                   65                             3180   
3                                  118                             3090   
4                                   -1                              391   
...                                ...                              ...   
581007                              17                              108   
581008                              12                               95   
581009                               7                               90   
581010                               5                               90   
581011                               4                               67   

        Hillshade_9am  Hillshade_Noon  Hillshade_3pm  \
0                 221             232            148   
1                 220             235            151   
2                 234             238            135   
3                 238             238            122   
4                 220             234            150   
...               ...             ...            ...   
581007            240             237            118   
581008            240             237            119   
581009            236             241            130   
581010            230             245            143   
581011            231             244            141   

        Horizontal_Distance_To_Fire_Points  ...  Soil_Type32  Soil_Type33  \
0                                     6279  ...            0            0   
1                                     6225  ...            0            0   
2                                     6121  ...            0            0   
3                                     6211  ...            0            0   
4                                     6172  ...            0            0   
...                                    ...  ...          ...          ...   
581007                                 837  ...            0            0   
581008                                 845  ...            0            0   
581009                                 854  ...            0            0   
581010                                 864  ...            0            0   
581011                                 875  ...            0            0   

        Soil_Type34  Soil_Type35  Soil_Type36  Soil_Type37  Soil_Type38  \
0                 0            0            0            0            0   
1                 0            0            0            0            0   
2                 0            0            0            0            0   
3                 0            0            0            0            0   
4                 0            0            0            0            0   
...             ...          ...          ...          ...          ...   
581007            0            0            0            0            0   
581008            0            0            0            0          

In [4]:
dataset['Cover_Type'] = dataset['Cover_Type'].replace(7, 0)

Análisis del Dataset

In [5]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 581012 entries, 0 to 581011
Data columns (total 55 columns):
 #   Column                              Non-Null Count   Dtype
---  ------                              --------------   -----
 0   Elevation                           581012 non-null  int64
 1   Aspect                              581012 non-null  int64
 2   Slope                               581012 non-null  int64
 3   Horizontal_Distance_To_Hydrology    581012 non-null  int64
 4   Vertical_Distance_To_Hydrology      581012 non-null  int64
 5   Horizontal_Distance_To_Roadways     581012 non-null  int64
 6   Hillshade_9am                       581012 non-null  int64
 7   Hillshade_Noon                      581012 non-null  int64
 8   Hillshade_3pm                       581012 non-null  int64
 9   Horizontal_Distance_To_Fire_Points  581012 non-null  int64
 10  Wilderness_Area1                    581012 non-null  int64
 11  Wilderness_Area2                    581012 non-null 

In [6]:
# hacemos uso del DataFrame llamado 'dataset' que contiene nuestros datos datos
# y es la columna que contiene las etiquetas de clase

#creamos una variable temporal que contentra toda la columna de 'Cover_Type'
y_temp = dataset['Cover_Type']

# Para la clase 0

#donde y_temp es igual a 0, separamos los datos en train_class_0, test_class_0
data_class_0 = dataset[y_temp == 0]
train_class_0, test_class_0 = train_test_split(data_class_0, test_size=0.2, random_state=42)

# Para la clase 1

#donde y_temp es igual a 1, separamos los datos en train_class_1, test_class_1
data_class_1 = dataset[y_temp == 1]
train_class_1, test_class_1 = train_test_split(data_class_1, test_size=0.2, random_state=42)

# Para la clase 2

data_class_2 = dataset[y_temp == 2]
train_class_2, test_class_2 = train_test_split(data_class_2, test_size=0.2, random_state=42)

# Para la clase e

data_class_3 = dataset[y_temp == 3]
train_class_3, test_class_3 = train_test_split(data_class_3, test_size=0.2, random_state=42)

# Para la clase 4

data_class_4 = dataset[y_temp == 4]
train_class_4, test_class_4 = train_test_split(data_class_4, test_size=0.2, random_state=42)

# Para la clase 5

data_class_5 = dataset[y_temp == 5]
train_class_5, test_class_5 = train_test_split(data_class_5, test_size=0.2, random_state=42)

# Para la clase 6

data_class_6 = dataset[y_temp == 6]
train_class_6, test_class_6 = train_test_split(data_class_6, test_size=0.2, random_state=42)

In [7]:
print(f"Para la clase 0 tenemos una cantidad de: { data_class_0.shape[0]} donde el 80% es: {train_class_0.shape[0]} y el 20% es: {test_class_0.shape[0]}")
print(f"Para la clase 1 tenemos una cantidad de: { data_class_1.shape[0]} donde el 80% es: {train_class_1.shape[0]} y el 20% es: {test_class_1.shape[0]}")
print(f"Para la clase 2 tenemos una cantidad de: { data_class_2.shape[0]} donde el 80% es: {train_class_2.shape[0]} y el 20% es: {test_class_2.shape[0]}")
print(f"Para la clase 3 tenemos una cantidad de: { data_class_3.shape[0]} donde el 80% es: {train_class_3.shape[0]} y el 20% es: {test_class_3.shape[0]}")
print(f"Para la clase 4 tenemos una cantidad de: { data_class_4.shape[0]} donde el 80% es: {train_class_4.shape[0]} y el 20% es: {test_class_4.shape[0]}")
print(f"Para la clase 5 tenemos una cantidad de: { data_class_5.shape[0]} donde el 80% es: {train_class_5.shape[0]} y el 20% es: {test_class_5.shape[0]}")
print(f"Para la clase 6 tenemos una cantidad de: { data_class_6.shape[0]} donde el 80% es: {train_class_6.shape[0]} y el 20% es: {test_class_6.shape[0]}")
print(f"La cantidad total de datos es: {dataset.shape[0]}")

Para la clase 0 tenemos una cantidad de: 20510 donde el 80% es: 16408 y el 20% es: 4102
Para la clase 1 tenemos una cantidad de: 211840 donde el 80% es: 169472 y el 20% es: 42368
Para la clase 2 tenemos una cantidad de: 283301 donde el 80% es: 226640 y el 20% es: 56661
Para la clase 3 tenemos una cantidad de: 35754 donde el 80% es: 28603 y el 20% es: 7151
Para la clase 4 tenemos una cantidad de: 2747 donde el 80% es: 2197 y el 20% es: 550
Para la clase 5 tenemos una cantidad de: 9493 donde el 80% es: 7594 y el 20% es: 1899
Para la clase 6 tenemos una cantidad de: 17367 donde el 80% es: 13893 y el 20% es: 3474
La cantidad total de datos es: 581012


Separación del 80% de datos para entrenamiento y 20% para pruebas

In [9]:
#separando los datos de entrenamiento y pruebas

#para los datos de entrenamiento
train_dataset = pd.concat([train_class_0, train_class_1, train_class_2, train_class_3, train_class_4, train_class_5, train_class_6]).values
test_dataset = pd.concat([test_class_0, test_class_1, test_class_2, test_class_3, test_class_4, test_class_5, test_class_6]).values


indices_train = np.random.permutation(len(train_dataset))
indices_test = np.random.permutation(len(test_dataset))

train_dataset = pd.concat([train_class_0, train_class_1, train_class_2, train_class_3, train_class_4, train_class_5, train_class_6]).values

train_dataset = train_dataset[indices_train]
train_test = test_dataset[indices_test]

m_train = len(train_dataset)
m_test = len(test_dataset)

In [10]:
print(m_train)
print(m_test)
print("El total de ejemplos es: ", (m_train + m_test))

464807
116205
El total de ejemplos es:  581012


Conversión de los datos a tensores de Pythorch

In [11]:
class TuConjuntoDatos(Dataset):
    def __init__(self, data, targets):
        self.data = data
        self.targets = targets

    def __len__(self):
        return len(self.data)

    def __getitem__(self, idx):
        features = torch.tensor(self.data[idx, :-1], dtype=torch.float32)  # Todas las columnas excepto la última
        target = torch.tensor(self.data[idx, -1], dtype=torch.long)  # Última columna (etiqueta)
        return features, target

    def get_features_shape(self):
        return (self.data.shape[1] - 1)

In [12]:
train_dataset = TuConjuntoDatos(train_dataset, -1)  # No se necesitan etiquetas durante el entrenamiento
test_dataset = TuConjuntoDatos(test_dataset, -1)

In [13]:
print(f"el 80% de datos es:{len(train_dataset)} y el 20% es: {len(test_dataset)}")
print( "la totalidad de datos es: ", dataset.shape[0])

el 80% de datos es:464807 y el 20% es: 116205
la totalidad de datos es:  581012


Creción de la clase de Red Neuronal MLS

In [14]:
class RedNeuronalMLS(nn.Module):
    def __init__(self, input_size, num_classes):
        super(RedNeuronalMLS, self).__init__()
        self.fc1 = nn.Linear(input_size, 50)
        self.fc2 = nn.Linear(50, num_classes)

    def forward(self, x):
        x = self.fc1(x)
        x = F.sigmoid(x)
        x = self.fc2(x)
        return x

Arquitectura del modelo



*   Capa de entrada, cantidad de características con las que cuenta el dataset
*   Capa de salida, cuenta con 7 clases


In [15]:
# Configurando parametros necesario
input_layer_size = train_dataset.get_features_shape() # Entrada de 54 caracteristicas, es la cantidad de columnas existente en el dataset
num_labels = 7         # como es una clasificacion tiene solo 2 salidas

print("la cantidad de elementos en la capa de entrada es: ",input_layer_size)
print("la cantidad de elementos en la capa de salida es: ",num_labels)

la cantidad de elementos en la capa de entrada es:  54
la cantidad de elementos en la capa de salida es:  7


Hiperparámetros

In [16]:
input_size = input_layer_size
num_classes = num_labels
learning_rate = 0.001
batch_size = 1000
num_epochs = 3

print("la cantidad de elementos en la entrada es: ",input_size)

la cantidad de elementos en la entrada es:  54


Verficación del GPU habilitado

In [17]:
# Configuramos el dispositivo cuda para GPU si está disponible
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
print(device)

cpu


Cargar conjunto de datos en lotes

In [18]:
print(len(train_dataset))
print(len(test_dataset))

train_loader = DataLoader(dataset=train_dataset, batch_size=batch_size, shuffle=True)
test_loader = DataLoader(dataset=test_dataset, batch_size=batch_size, shuffle=True)

print("la cantidad de lotes para el dataset de entrenamiento es: ",len(train_loader))
print("la cantidad de lotes para el dataset de prueba es: ",len(test_loader))

464807
116205
la cantidad de lotes para el dataset de entrenamiento es:  465
la cantidad de lotes para el dataset de prueba es:  117


Modelo Red Nueronal

In [19]:
# Initialize network
model = RedNeuronalMLS(input_size=input_size, num_classes=num_classes).to(device)

In [20]:
model

RedNeuronalMLS(
  (fc1): Linear(in_features=54, out_features=50, bias=True)
  (fc2): Linear(in_features=50, out_features=7, bias=True)
)

Función de pérdida o costo para el modelo

In [21]:
# Perdida y optimizador
criterion = nn.CrossEntropyLoss()
optimizer = optim.Adam(model.parameters(), lr=learning_rate)

Entrenamiento de la Red Neuronal

In [22]:
# Train Network
for epoch in range(num_epochs):
    for batch_idx, (data, targets) in enumerate(tqdm(train_loader)):

        # Obtenemos el CUDA si es posible
        data = data.to(device=device)
        targets = targets.to(device=device)


        # Llegar a la forma correcta
        data = data.reshape(data.shape[0], -1)
        print(data.shape)
        # print("-"*30)

        # forward
        scores = model(data)
        loss = criterion(scores, targets)

        # backward
        optimizer.zero_grad()
        loss.backward()

        # gradient descent or adam step
        optimizer.step()

  0%|          | 1/465 [00:00<02:16,  3.41it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])


  1%|          | 5/465 [00:00<00:37, 12.11it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


  2%|▏         | 11/465 [00:00<00:23, 18.98it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


  4%|▍         | 20/465 [00:00<00:15, 28.62it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


  6%|▌         | 28/465 [00:01<00:13, 33.16it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


  8%|▊         | 37/465 [00:01<00:11, 36.97it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 10%|▉         | 45/465 [00:01<00:15, 26.51it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 11%|█▏        | 53/465 [00:02<00:13, 30.79it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 12%|█▏        | 58/465 [00:02<00:12, 33.86it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 15%|█▍        | 68/465 [00:02<00:17, 23.09it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 17%|█▋        | 77/465 [00:03<00:13, 28.72it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 19%|█▊        | 87/465 [00:03<00:10, 34.67it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 21%|██        | 96/465 [00:03<00:09, 37.56it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 23%|██▎       | 105/465 [00:03<00:09, 39.62it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 24%|██▎       | 110/465 [00:03<00:09, 37.97it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 26%|██▌       | 119/465 [00:04<00:09, 36.44it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 28%|██▊       | 128/465 [00:04<00:08, 38.07it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 29%|██▉       | 136/465 [00:04<00:08, 38.14it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 31%|███       | 144/465 [00:04<00:08, 37.72it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 33%|███▎      | 152/465 [00:05<00:14, 22.24it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 34%|███▍      | 160/465 [00:05<00:11, 27.41it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 36%|███▌      | 168/465 [00:05<00:09, 30.98it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 38%|███▊      | 176/465 [00:05<00:08, 33.12it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 39%|███▊      | 180/465 [00:06<00:08, 32.23it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 40%|████      | 188/465 [00:06<00:08, 33.59it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 42%|████▏     | 196/465 [00:06<00:07, 35.15it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 44%|████▍     | 204/465 [00:06<00:07, 36.20it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 46%|████▌     | 212/465 [00:07<00:06, 36.76it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 47%|████▋     | 220/465 [00:07<00:07, 34.31it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 49%|████▉     | 228/465 [00:07<00:06, 34.51it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 51%|█████     | 236/465 [00:08<00:10, 22.15it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 52%|█████▏    | 244/465 [00:08<00:08, 26.38it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 54%|█████▍    | 252/465 [00:08<00:07, 29.44it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 56%|█████▌    | 260/465 [00:08<00:06, 32.04it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 58%|█████▊    | 269/465 [00:08<00:05, 36.64it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 60%|█████▉    | 278/465 [00:09<00:04, 38.21it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 62%|██████▏   | 286/465 [00:09<00:04, 37.09it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 63%|██████▎   | 291/465 [00:09<00:04, 38.58it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 65%|██████▍   | 301/465 [00:09<00:04, 40.25it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 67%|██████▋   | 311/465 [00:10<00:03, 40.27it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 69%|██████▉   | 321/465 [00:10<00:05, 27.87it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 71%|███████   | 330/465 [00:10<00:04, 32.70it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 72%|███████▏  | 334/465 [00:10<00:04, 29.13it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 73%|███████▎  | 341/465 [00:11<00:04, 25.62it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 74%|███████▍  | 344/465 [00:11<00:05, 24.03it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 75%|███████▌  | 350/465 [00:11<00:04, 23.37it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 77%|███████▋  | 356/465 [00:11<00:04, 23.08it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 78%|███████▊  | 362/465 [00:12<00:04, 24.89it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 78%|███████▊  | 365/465 [00:12<00:04, 23.67it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 79%|███████▉  | 368/465 [00:12<00:04, 21.57it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 80%|████████  | 374/465 [00:12<00:04, 21.55it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 82%|████████▏ | 380/465 [00:12<00:03, 23.51it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 83%|████████▎ | 386/465 [00:13<00:03, 24.91it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 84%|████████▍ | 392/465 [00:13<00:02, 25.72it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 86%|████████▌ | 398/465 [00:13<00:02, 26.33it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 87%|████████▋ | 404/465 [00:14<00:03, 16.92it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 88%|████████▊ | 407/465 [00:14<00:03, 18.04it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 89%|████████▉ | 413/465 [00:14<00:02, 20.24it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 90%|█████████ | 419/465 [00:14<00:02, 20.78it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 92%|█████████▏| 427/465 [00:15<00:01, 28.19it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 93%|█████████▎| 431/465 [00:15<00:01, 29.34it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 94%|█████████▍| 439/465 [00:15<00:00, 30.82it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 96%|█████████▌| 446/465 [00:15<00:00, 29.43it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 98%|█████████▊| 454/465 [00:15<00:00, 31.43it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 99%|█████████▉| 462/465 [00:16<00:00, 32.78it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


100%|██████████| 465/465 [00:16<00:00, 28.50it/s]


torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([807, 54])


  1%|▏         | 6/465 [00:00<00:16, 27.01it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


  3%|▎         | 14/465 [00:00<00:14, 31.93it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


  5%|▍         | 22/465 [00:00<00:12, 34.69it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


  6%|▋         | 30/465 [00:00<00:12, 34.37it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


  8%|▊         | 38/465 [00:01<00:12, 34.97it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


  9%|▉         | 42/465 [00:01<00:11, 35.68it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 12%|█▏        | 54/465 [00:01<00:11, 35.92it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 13%|█▎        | 62/465 [00:01<00:11, 35.38it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 14%|█▍        | 66/465 [00:01<00:11, 34.64it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 16%|█▌        | 74/465 [00:02<00:17, 22.39it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 18%|█▊        | 82/465 [00:02<00:13, 28.57it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 20%|██        | 94/465 [00:03<00:10, 34.22it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 22%|██▏       | 103/465 [00:03<00:09, 37.24it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 23%|██▎       | 108/465 [00:03<00:09, 37.68it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 26%|██▌       | 121/465 [00:03<00:08, 39.26it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 28%|██▊       | 129/465 [00:03<00:08, 37.39it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 29%|██▉       | 137/465 [00:04<00:08, 36.59it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 31%|███       | 145/465 [00:04<00:08, 36.54it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 33%|███▎      | 153/465 [00:04<00:08, 35.04it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 35%|███▍      | 162/465 [00:04<00:08, 37.33it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 37%|███▋      | 171/465 [00:05<00:07, 38.57it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 39%|███▊      | 180/465 [00:05<00:07, 39.89it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 40%|███▉      | 185/465 [00:05<00:06, 40.26it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 42%|████▏     | 195/465 [00:05<00:06, 40.00it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 44%|████▍     | 205/465 [00:05<00:06, 39.99it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 46%|████▌     | 215/465 [00:06<00:06, 40.69it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 48%|████▊     | 224/465 [00:06<00:08, 27.70it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 50%|████▉     | 232/465 [00:06<00:07, 30.90it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 52%|█████▏    | 240/465 [00:07<00:07, 32.05it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 53%|█████▎    | 248/465 [00:07<00:06, 33.84it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 55%|█████▌    | 256/465 [00:07<00:06, 33.05it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 57%|█████▋    | 264/465 [00:07<00:05, 34.52it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 58%|█████▊    | 268/465 [00:07<00:05, 33.72it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 59%|█████▉    | 276/465 [00:08<00:05, 34.33it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 61%|██████    | 284/465 [00:08<00:05, 35.57it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 63%|██████▎   | 292/465 [00:08<00:05, 32.68it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 64%|██████▎   | 296/465 [00:08<00:05, 28.80it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 65%|██████▍   | 302/465 [00:09<00:06, 25.01it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 66%|██████▌   | 308/465 [00:09<00:06, 24.42it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 67%|██████▋   | 311/465 [00:09<00:06, 24.71it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 69%|██████▉   | 320/465 [00:09<00:05, 25.11it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 70%|███████   | 326/465 [00:10<00:08, 17.11it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 71%|███████▏  | 332/465 [00:10<00:06, 21.03it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 73%|███████▎  | 338/465 [00:10<00:05, 22.80it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 73%|███████▎  | 341/465 [00:10<00:05, 22.50it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 75%|███████▍  | 347/465 [00:11<00:04, 23.72it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 76%|███████▌  | 353/465 [00:11<00:04, 24.95it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 77%|███████▋  | 359/465 [00:11<00:04, 25.55it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 78%|███████▊  | 365/465 [00:11<00:04, 24.38it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 79%|███████▉  | 368/465 [00:11<00:03, 24.53it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 80%|████████  | 374/465 [00:12<00:03, 22.89it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 82%|████████▏ | 380/465 [00:12<00:03, 23.35it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 83%|████████▎ | 386/465 [00:12<00:03, 24.58it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 85%|████████▍ | 394/465 [00:12<00:02, 30.84it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 86%|████████▋ | 402/465 [00:13<00:01, 34.68it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 87%|████████▋ | 406/465 [00:13<00:01, 35.18it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 88%|████████▊ | 410/465 [00:13<00:02, 22.61it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 90%|█████████ | 419/465 [00:13<00:01, 29.11it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 92%|█████████▏| 427/465 [00:14<00:01, 29.80it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 94%|█████████▎| 435/465 [00:14<00:00, 32.25it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 95%|█████████▌| 443/465 [00:14<00:00, 33.80it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 97%|█████████▋| 451/465 [00:14<00:00, 34.01it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 98%|█████████▊| 455/465 [00:14<00:00, 33.56it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


100%|█████████▉| 463/465 [00:15<00:00, 33.96it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


100%|██████████| 465/465 [00:15<00:00, 30.45it/s]


torch.Size([807, 54])


  0%|          | 2/465 [00:00<00:28, 16.46it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


  1%|▏         | 6/465 [00:00<00:16, 27.39it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


  2%|▏         | 10/465 [00:00<00:14, 30.60it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


  3%|▎         | 14/465 [00:00<00:13, 32.49it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


  4%|▍         | 18/465 [00:00<00:13, 32.30it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


  5%|▍         | 22/465 [00:00<00:13, 33.10it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


  6%|▌         | 26/465 [00:00<00:12, 34.30it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


  6%|▋         | 30/465 [00:00<00:12, 34.76it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


  7%|▋         | 34/465 [00:01<00:12, 35.57it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


  8%|▊         | 38/465 [00:01<00:11, 36.76it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


  9%|▉         | 42/465 [00:01<00:11, 37.20it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 10%|▉         | 46/465 [00:01<00:11, 37.23it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 11%|█         | 50/465 [00:01<00:10, 37.90it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])

 12%|█▏        | 54/465 [00:01<00:10, 38.08it/s]


torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 12%|█▏        | 58/465 [00:01<00:11, 36.57it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 13%|█▎        | 62/465 [00:01<00:11, 35.09it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 15%|█▌        | 70/465 [00:02<00:11, 35.11it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 16%|█▌        | 74/465 [00:02<00:11, 34.36it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 17%|█▋        | 78/465 [00:02<00:11, 34.75it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 18%|█▊        | 82/465 [00:02<00:11, 34.47it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 18%|█▊        | 86/465 [00:02<00:11, 34.02it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 19%|█▉        | 90/465 [00:02<00:11, 33.60it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 21%|██        | 98/465 [00:03<00:16, 21.68it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 23%|██▎       | 106/465 [00:03<00:13, 27.01it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 25%|██▍       | 114/465 [00:03<00:11, 29.56it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 25%|██▌       | 118/465 [00:03<00:12, 28.25it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 27%|██▋       | 127/465 [00:04<00:10, 33.71it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 29%|██▉       | 136/465 [00:04<00:09, 36.56it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 31%|███       | 145/465 [00:04<00:08, 38.31it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 33%|███▎      | 153/465 [00:04<00:08, 38.43it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 35%|███▍      | 162/465 [00:04<00:07, 38.47it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 37%|███▋      | 172/465 [00:05<00:07, 39.96it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 39%|███▉      | 181/465 [00:05<00:07, 40.30it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 40%|████      | 186/465 [00:05<00:07, 39.27it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 42%|████▏     | 195/465 [00:05<00:07, 37.96it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 44%|████▎     | 203/465 [00:05<00:06, 38.39it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 46%|████▌     | 212/465 [00:06<00:07, 32.93it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 47%|████▋     | 220/465 [00:06<00:10, 24.36it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 49%|████▉     | 228/465 [00:06<00:08, 28.49it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 51%|█████     | 237/465 [00:07<00:07, 31.65it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 52%|█████▏    | 241/465 [00:07<00:12, 18.14it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])


 53%|█████▎    | 247/465 [00:08<00:14, 15.29it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 54%|█████▍    | 253/465 [00:08<00:10, 19.31it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 56%|█████▌    | 259/465 [00:08<00:09, 22.70it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 57%|█████▋    | 265/465 [00:08<00:08, 24.21it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 58%|█████▊    | 271/465 [00:09<00:07, 25.27it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 60%|█████▉    | 277/465 [00:09<00:07, 25.68it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 61%|██████    | 283/465 [00:09<00:06, 26.95it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 62%|██████▏   | 289/465 [00:09<00:06, 26.69it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 63%|██████▎   | 295/465 [00:10<00:06, 25.77it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 64%|██████▍   | 298/465 [00:10<00:06, 25.64it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 65%|██████▌   | 304/465 [00:10<00:06, 24.70it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 67%|██████▋   | 310/465 [00:10<00:06, 24.61it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 68%|██████▊   | 316/465 [00:10<00:05, 24.93it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 69%|██████▉   | 322/465 [00:11<00:05, 24.53it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 71%|███████   | 328/465 [00:11<00:05, 25.17it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 72%|███████▏  | 334/465 [00:11<00:05, 25.33it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 74%|███████▎  | 342/465 [00:11<00:04, 30.05it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 75%|███████▌  | 349/465 [00:12<00:05, 20.35it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 77%|███████▋  | 357/465 [00:12<00:04, 26.17it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 78%|███████▊  | 365/465 [00:12<00:03, 29.65it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 80%|████████  | 373/465 [00:13<00:02, 32.55it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 81%|████████  | 377/465 [00:13<00:02, 31.93it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 83%|████████▎ | 385/465 [00:13<00:02, 34.58it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 85%|████████▍ | 393/465 [00:13<00:02, 35.15it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 86%|████████▌ | 401/465 [00:13<00:01, 33.89it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 88%|████████▊ | 409/465 [00:14<00:01, 33.44it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 90%|████████▉ | 417/465 [00:14<00:01, 33.87it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 91%|█████████▏| 425/465 [00:14<00:01, 34.22it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 93%|█████████▎| 433/465 [00:15<00:01, 21.88it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])

 95%|█████████▍| 441/465 [00:15<00:00, 26.50it/s]


torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 97%|█████████▋| 449/465 [00:15<00:00, 30.73it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


 98%|█████████▊| 457/465 [00:15<00:00, 32.32it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])


100%|██████████| 465/465 [00:16<00:00, 28.91it/s]

torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([1000, 54])
torch.Size([807, 54])


Validaciones

In [23]:
# Check accuracy on training & test to see how good our model
def check_accuracy(loader, model):
    num_correct = 0
    num_samples = 0
    model.eval()

    predicciones = []
    with torch.no_grad():
        for x, y in loader:
            x = x.to(device=device)
            y = y.to(device=device)
            x = x.reshape(x.shape[0], -1)

            scores = model(x)
            _, predictions = scores.max(1)
            predicciones.append(predictions)

            num_correct += (predictions == y).sum()
            num_samples += predictions.size(0)

    model.train()
    return num_correct/num_samples, predicciones

Precisión con los datos de entrenamiento

In [24]:
p_train, pred_train  = check_accuracy(train_loader, model)
print(f"Precisión en el conjunto de entrenamiento: {p_train*100:.2f} %")

Precisión en el conjunto de entrenamiento: 68.72 %


Precisión con los datos de prueba

In [25]:
p_test, pred_test  = check_accuracy(test_loader, model)
print(f"Precisión en el conjunto de prueba: {p_test*100:.2f} %")

Precisión en el conjunto de prueba: 68.71 %


Se puede determinar que el modelo reliza las predicciones con una precisión del 68.71 % con los datos de prueba y con un 68.72 % con los datos de entrenamiento. Tener el modelo con esta precision puede llegar a generar desconfianza.